x/evm/types/state_transition.go

```diff
@@ -168,18 +168,13 @@ func (st StateTransition) TransitionDb(ctx sdk.Context, config ChainConfig) (exe
 		to = EthAddressToString(st.Recipient)
 		recipientStr = to
 	}
-	enableDebug := checkTracesSegment(ctx.BlockHeight(), senderStr, to)
-
-	var tracer vm.Tracer
-	if st.TraceTxLog || enableDebug {
-		tracer = vm.NewStructLogger(evmLogConfig)
-	} else {
-		tracer = NewNoOpTracer()
+	tracer, err := newTracer(ctx, st.TxHash)
+	if err != nil {
+		return
```

Comment:
it's shouldn't return even in error, because that will change the code logic, it's a risk to occur smb

Reply:
return NoOpTracer() if err